pkg/userdata/coreos/userdata.go

```diff
@@ -21,15 +21,15 @@ type config struct {
 }
 
 func getConfig(r runtime.RawExtension) (*config, error) {
-	p := config{}
+	p := new(config)
```

Comment:
I'd really discourage you from this. New hides the fact that a points is created. If you want to do this then make it explicitly: `p := &config{}`.

Reply:
`new` has a defined behaviour, it always returns a value of type `*T`, from my point of view, it's way more clearer than `p := &config{}`

Reply:
Both are valid solutions

Reply:
Yes. Both are valid solutions. The `&config{}` one seems to be used by more people in the community. To summarize this PR: I wouldn't create a pointer in this func right away, especially due to the fact that we can more explicitly pass the pointer to `json.Unmarshal` :relaxed: